<?xml version="1.0" encoding="utf-8"?>
<launch>
    <!-- start basic args -->
    <arg name="veh" default="default"/>
    <arg name="param_file_name" default="default" />
    <arg name="verbose" default="true" />
    <arg name="loc" default="-1"/>
    <arg name="line_detector_param_file_name" default="$(arg param_file_name)"/>

    <!-- Benchmark path -->
    <group>
        <remap from="static_controller_node/wheels_cmd_executed" to="wheels_driver_node/wheels_cmd_executed" />
        <remap from="static_controller_node/car_cmd" to="lane_controller_node/car_cmd" />
        <remap from="static_controller_node/actuator_limits" to="inverse_kinematics_node/actuator_limits" />
        <remap from="static_controller_node/reset_env" to="reset_env"/>
        <remap from="static_controller_node/pose2d" to="pose2d"/>
        <arg name="pkg_name" default="static_controller" doc="name of the package"/>
        <arg name="node_name" default="static_controller_node" doc="name of the node"/>

        <group ns="$(arg veh)">
            <node  pkg="$(arg pkg_name)" type="$(arg node_name).py" name="$(arg node_name)" output="screen" required="true"></node>
        </group>
    </group>

    <!-- Kinematics: turns chassis-level commands into wheels-level commands and chassis-level velocities -->
    <remap from="kinematics_node/car_cmd" to="lane_controller_node/car_cmd"/>
    <remap from="kinematics_node/wheels_cmd" to="wheels_driver_node/wheels_cmd"/>
    <include file="$(find dagu_car)/launch/kinematics_node.launch">
        <arg name="veh" value="$(arg veh)"/>
        <arg name="param_file_name" value="$(arg param_file_name)"/>
    </include>

    
    <!-- Start Test Node -->
<!--         <node pkg="dt_dependent_node" name="dt_dependent_node" type="dt_dependent_node.py" output="screen" />  -->
    <!-- End Test Node -->
</launch>
